foxpuppet/windows.py

```diff
@@ -2,44 +2,50 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
-from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
+        self.windows = None
+
+        self.populate_windows()
+
+    def populate_windows(self):
+        self.windows = []
+        for handle in self.selenium.window_handles:
+            self.windows.append(BaseWindow(self.selenium, handle))
+
+
+class BaseWindow(object):
+
+    def __init__(self, selenium, *args, **kwargs):
+        self.selenium = selenium
         self._handle = None
 
     @property
-    def all(self):
-        """Returns all current window handles"""
-        return self.selenium.window_handles
+    def handle(self):
+        return self._handle
+
+    @handle.setter
+    def handle(self, value):
+        self._handle = value
 
     @property
-    def current(self):
-        """Returns current window handle"""
-        return self.selenium.current_window_handle
+    def wait(self):
+        return WebDriverWait(self.selenium, 10)
 
     def focus(self, handle):
         """Switch Selenium focus to a specific window"""
-        return self.selenium.switch_to.window(handle)
 
-    @contextmanager
-    def wait_for_new_window(self):
-        handles_before = self.selenium.window_handles
-        yield
-        WebDriverWait(self.selenium, 10).until(
-            lambda driver: len(handles_before) != len(driver.window_handles))
-
-    @property
-    def window_element(self):
-        """Returns the inner DOM window element.
+        self.selenium.switch_to.window(self.handle)
 
-        :returns: DOM window element.
-        """
+        with self.selenium.context('chrome'):
+            self.selenium.execute_script('window.focus();')
 
-        return self.selenium.find_element(By.CSS_SELECTOR, ':root')
```

Comment:
I dont really understand why this line and others have been moved to the browser window class.